finetrainers/trainer.py

```diff
@@ -73,35 +80,93 @@ def __init__(self, args: Args) -> None:
         # Autoencoders
         self.vae = None
 
+        # Scheduler
+        self.scheduler = None
+
         self._init_distributed()
         self._init_logging()
         self._init_directories_and_repositories()
 
         self.state.model_name = self.args.model_name
         self.model_config = get_config_from_model_name(self.args.model_name, self.args.training_type)
 
-    def prepare_models(self) -> None:
-        logger.info("Initializing models")
+    def prepare_dataset(self) -> None:
+        # TODO(aryan): Make a background process for fetching
```

Comment:
Is this `prefetch_factor` we're talking about? If not, maybe a more elaborate comment would be better.

Reply:
Currently, loading the dataset is a blocking process. We load one thing and process that thing and repeat. Prefetching here with any factor does not really help. We don't get the desired effect because the main process will still be slowed down due to loading and dispatching. The process of loading from disk to memory and converting to tensor, if done on a separate background thread is faster, and better overlaps for overlapping with computation from my preliminary tests.

Reply:
Interesting. Definitely eager to see it.